hub.go

```diff
@@ -202,24 +219,25 @@ type jwtConfig struct {
 //
 // If you change this, also update the Caddy module and the documentation.
 type opt struct {
-	transport          Transport
-	topicSelectorStore *TopicSelectorStore
-	anonymous          bool
-	debug              bool
-	subscriptions      bool
-	ui                 bool
-	demo               bool
-	logger             Logger
-	writeTimeout       time.Duration
-	dispatchTimeout    time.Duration
-	heartbeat          time.Duration
-	publisherJWT       *jwtConfig
-	subscriberJWT      *jwtConfig
-	metrics            Metrics
-	allowedHosts       []string
-	publishOrigins     []string
-	corsOrigins        []string
-	cookieName         string
+	transport                    Transport
+	topicSelectorStore           *TopicSelectorStore
+	anonymous                    bool
+	debug                        bool
+	subscriptions                bool
+	ui                           bool
+	demo                         bool
+	logger                       Logger
+	writeTimeout                 time.Duration
+	dispatchTimeout              time.Duration
+	heartbeat                    time.Duration
+	publisherJWT                 *jwtConfig
+	subscriberJWT                *jwtConfig
+	metrics                      Metrics
+	allowedHosts                 []string
+	publishOrigins               []string
+	corsOrigins                  []string
+	cookieName                   string
+	protocolVersionCompatibility int
```

Comment:
I think it would a good idea to introduce `opt.IsProtocolVersionCompatibleWith(x)` to avoid repeating `if opt.protocolVersionCompatibility == 0 || opt.protocolVersionCompatibility >= 8` in the future ( it would be `opt.IsProtocolVersionCompatibleWith(7)` ).

Reply:
Done